python/lsst/dax/obscore/siav2.py

```diff
@@ -0,0 +1,703 @@
+                extra_dims = {"visit"}
+            else:
+                return None
+
+        if not regions:
+            # In case we are called with no regions.
+            return WhereBind(where="")
+
+        # Must OR together all regions. We can do this in sphgeom directly.
+        # Butler query system itself can not OR regions.
+        region = regions[0]
+        if len(regions) > 1:
+            for r in regions[1:]:
+                region = UnionRegion(region, r)
+        return WhereBind(
+            where=f"{region_dim}.region OVERLAPS(region)",
+            bind={"region": region},
+            extra_dims=extra_dims,
+        )
+
+    def from_time(
+        self, ts: Iterable[astropy.time.Time | Timespan], dimensions: DimensionGroup
+    ) -> WhereBind | None:
+        """Convert a time span to a butler where clause.
+
+        Parameters
+        ----------
+        ts : `~collections.abc.Iterable` of \
+                [ `astropy.time.Time` | `lsst.daf.butler.Timespan` ]
+            The times of interest.
+        dimensions : `lsst.daf.butler.DimensionGroup`
+            The dimensions for the dataset type being queried.
+
+        Returns
+        -------
+        where : `WhereBind` or `None`
+            The where clause or `None` if time is not supported by this
+            dataset type.
+        """
+        time_dim = dimensions.timespan_dimension
+        if not time_dim:
+            return None
+        wheres = []
+        for i, t in enumerate(ts):
+            wheres.append(WhereBind(where=f"{time_dim}.timespan OVERLAPS(ts{i})", bind={f"ts{i}": t}))
+
+        return WhereBind.combine(wheres, mode="OR")
+
+    def from_exptime(
+        self, exptime_intervals: Iterable[Interval], dimensions: DimensionGroup
+    ) -> WhereBind | None:
+        """Convert an exposure time interval to a butler where clause.
+
+        Parameters
+        ----------
+        exptime_intervals : `~collections.abc.Iterable` [ `Interval` ]
+            The exposure time interval of interest as two floating point UTC
+            MJDs.
+        dimensions : `lsst.daf.butler.DimensionGroup`
+            The dimensions for the dataset type being queried.
+
+        Returns
+        -------
+        wheres : `WhereBind` or `None`
+            The where clause for the exposure times, or `None` if the dataset
+            type does not understand exposure time.
+        """
+        if "exposure" in dimensions:
+            exp_dim = "exposure"
+        elif "visit" in dimensions:
+            exp_dim = "visit"
+        else:
+            return None
+        wheres = []
+        for exptime_interval in exptime_intervals:
+            exp_wheres = []
+            for cmp, value in zip((">", "<"), exptime_interval, strict=True):
+                if math.isfinite(value):
+                    exp_wheres.append(WhereBind(where=f"{exp_dim}.exposure_time {cmp} {value}"))
+            wheres.append(WhereBind.combine(exp_wheres))
```

Comment:
If the interval is `(-Inf, +Inf)`, will this produce something reasonable?

Reply:
Well spotted. Does indeed blow up.